File: src/swift/fibonacciIterative.swift
// Function to calculate Fibonacci numbers iteratively
func fibonacci(_ number: Int) -> Int {
    if number <= 1 {
        return number
    } else {
        var currentNumber: Int = 1
        var nextNumber: Int = 0

        // Iterate from 0 to 'number - 1'
        for _ in 0..<number {
            let tempNumber = nextNumber
            nextNumber = nextNumber + currentNumber
            currentNumber = tempNumber
        }

        return currentNumber
    }
}

let index: Int = 15
print("Fibonacci (iterative):", fibonacci(index))

File: src/rust/fibonacci_iterative.rs
fn fibonacci(n: u32) -> u64 {
    if n <= 1 {
        return n as u64;
    }

    let mut a: u64 = 0;
    let mut b: u64 = 1;

    for _ in 2..=n {
        let temp = a;
        a = b;
        b += temp;
    }

    b
}

fn main() {
    let index = 15;
    println!("Fibonacci (iterative) of {} is: {}", index, fibonacci(index));
}
